src/app/modules/common/dialog/job-progress/job-progress-dialog.component.ts

```diff
@@ -0,0 +1,167 @@
+import { AfterViewChecked, Component, Inject, OnInit } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogConfig, MatDialogRef } from '@angular/material/dialog';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import _ from 'lodash';
+import { Observable, Subscription, delay, map } from 'rxjs';
+import { JobState } from 'app/enums/job-state.enum';
+import { Job, JobProgress } from 'app/interfaces/job.interface';
+import { WebSocketService } from 'app/services/ws.service';
+
+export interface JobProgressDialogConfig {
+  job$: Observable<Job>;
+  callbacks: {
+    onSuccess?: (job: Job) => void;
+    onFailure?: (job: Job) => void;
+    onAbort?: (job: Job) => void;
+    onProgress: (progress: JobProgress) => void;
+  };
+  config: {
+    title: string;
+    description: string;
+  };
+  flags: {
+    showRealtimeLogs: boolean;
```

Comment:
realtime logs are only used in one form, I'd consider creating a separate component for this.